src/FormControl/FormControl.tsx

```diff
@@ -0,0 +1,232 @@
+import React from 'react'
+import {Autocomplete, Box, Checkbox, Radio, Select, Textarea, TextInput, TextInputWithTokens, useSSRSafeId} from '..'
+import InputValidation from '../_InputValidation'
+import {ComponentProps} from '../utils/types'
+import FormControlCaption from '../InputField/_InputFieldCaption'
+import FormControlLabel from '../InputField/_InputFieldLabel'
+import FormControlValidation from './_FormControlValidation'
+import FormControlLeadingVisual from './_FormControlLeadingVisual'
+import ValidationAnimationContainer from '../_ValidationAnimationContainer'
+import {Slots} from './slots'
+import {get} from '../constants'
+
+export interface Props {
+  children?: React.ReactNode
+  /**
+   * Whether the field is ready for user input
+   */
+  disabled?: boolean
+  /**
+   * The unique identifier for this field. Used to associate the label, validation text, and caption text
+   */
+  id?: string
+  /**
+   * Whether this field must have a value for the user to complete their task
+   */
+  required?: boolean
+  /**
+   * Changes the layout of the form control
+   */
+  variant?: 'stack' | 'choice' // TODO: come up with a better name for 'stack'
```

Comment:
Should we just go with "stack" for the vertically stacked label+input+caption/validation? Or does anybody have an idea for a better name?

Reply:
Could it be more abstract like `vertical` and `horizontal` with prop called `alignment` instead?

Taking that further, do we need a variant prop at all?

E.g. 

```jsx
      <FormControl variant="stack">
        <FormControl.Label>Checkbox option one</FormControl.Label>
        <Checkbox />
      </FormControl>

```

☝️ This is incorrect and breaks, but is the default value that user needs to manually change to `choice`. 

Can we read children via `React.Children` to see if there's a type of `Checkbox` or `Radio`, which automatically applies `choice` for them without needing it via prop?

Reply:
@rezrah We discussed that after you left one of our meetings. We decided that reading from `React.Children` felt like a little too much magic and would prevent people from using custom checkboxes/radios

Reply:
Thanks @colebemis. Should the values be more explicit to the type of child instead then? It's not immediately clear (to me at least) which values (stacked/choice) to use for `variant`, even though there are restrictions on types of input to variant. E.g. You can't use stacked with checkbox children.

E.g.
```jsx
      <FormControl variant="checkbox">
        <FormControl.Label>Checkbox option one</FormControl.Label>
        <Checkbox />
      </FormControl>
```

I personally find `for` even more readable...

```jsx
      <FormControl for="checkbox">
        <FormControl.Label>Checkbox option one</FormControl.Label>
        <Checkbox />
      </FormControl>
```

Reply:
@colebemis - I went with Reza's suggestion to remove `variant`. I won't merge this until you either say this is ok, or we come up with a solution everybody is comfortable with.

Reply:
> Given this component only supports internal form inputs, I think it makes sense to remove variant.

When we were looking at the API, we talked about supporting custom form inputs and that's where an explicit layout `variant` came from, because there is no non-hacky way for the user to get a vertical layout on their own.

If we are choosing to not support custom input elements for now, we can remove the variant prop. I'm always up for a little magic. I imagine the need for custom inputs will come up at some point and then we can bring back the `variant` prop, it's nice that we already have a solution.